src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -3527,7 +3543,7 @@ private BuildingRequestEventArgs FireBuildingRequest(BuildingRequestEventArgs bu
         /// <param name="options">options as specified by the user.</param>
         private void ValidateSaveChangesOptions(SaveChangesOptions options)
         {
-            const SaveChangesOptions All = SaveChangesOptions.ContinueOnError | SaveChangesOptions.BatchWithSingleChangeset | SaveChangesOptions.BatchWithIndependentOperations | SaveChangesOptions.ReplaceOnUpdate | SaveChangesOptions.PostOnlySetProperties | SaveChangesOptions.UseRelativeUri | SaveChangesOptions.UseJsonBatch;
+            const SaveChangesOptions All = SaveChangesOptions.ContinueOnError | SaveChangesOptions.BatchWithSingleChangeset | SaveChangesOptions.BatchWithIndependentOperations | SaveChangesOptions.ReplaceOnUpdate | SaveChangesOptions.PostOnlySetProperties | SaveChangesOptions.UseRelativeUri | SaveChangesOptions.UseJsonBatch | SaveChangesOptions.BulkUpdate;

```

Comment:
I don't think this code path will ever be reached..because for people to use bulk update they have to call the `DeepUpdate` method and there is no way they can combine the two. So even if they do this: `context.SaveChanges(SaveChangesOptions.BatchWithSingleChangeset | SaveChangesOptions.BulkUpdate);` nothing will happen technically.

Reply:
I agree nothing will happen technically, but it will better if we validate that and customers know that we are not allowed to combine `SaveChangesOptions.BulkUpdate` with any other option.

Reply:
Shouldn't this be made clear in the documentation of the feature? Let me just add